lib/galaxy/config/sample/datatypes_conf.xml.sample

```diff
@@ -558,6 +558,8 @@
     <!-- Excel datatypes -->
     <datatype extension="excel.xls" type="galaxy.datatypes.binary:ExcelXls" display_in_upload="true"/>
     <datatype extension="xlsx" type="galaxy.datatypes.binary:Xlsx" display_in_upload="true"/>
+    <!-- Word datatypes -->
+    <datatype extension="docx" type="galaxy.datatypes.binary:docx" display_in_upload="true"/>
```

Comment:
There is no pzthon class that actually implements it, isn't it? So you need to subclass the binary class